google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugGlobalPollerTimerTask.java

```diff
@@ -7,15 +9,18 @@
  */
 class CloudDebugGlobalPollerTimerTask extends TimerTask {
 
+  private final CloudDebugProcessStateCollector stateCollector;
   private CloudDebugGlobalPoller cloudDebugGlobalPoller;
 
   public CloudDebugGlobalPollerTimerTask(CloudDebugGlobalPoller cloudDebugGlobalPoller) {
     this.cloudDebugGlobalPoller = cloudDebugGlobalPoller;
+    stateCollector = ApplicationManager.getApplication()
```

Comment:
Yeah, that's a pretty frequent pattern. Luckily this way mocking these services is quite easy just by registering mocks into the application.